sdk/how-to-guides/ports.md

```diff
@@ -0,0 +1,144 @@
+> See first: [Juju `open-port`](https://juju.is/docs/juju/hook-tool#open-port)
+
```

Comment:
> Might be good to add another > line here that says:
> 
> > Added in Juju 3.1 and Ops x.x
> 
> Maybe also a note that this is only supported for machine charms.

It's supported for both machine and Kubernetes. I'm not sure when it was added for machines (before 2.9), but it's been available in K8s since Juju 3.1. `open_port`, `close_port` and `opened_ports` have been in ops since 2.1, and `set_ports` since 2.7.

It seems like too much detail to include all that in the how-to guide. The linked doc already says that Juju 3.1 is needed for using this with a Kubernetes charm, and I don't think it's a good idea to repeat that here.

Reply:
All things considered: Let's refine the note to say:

> For Kubernetes, support added in Juju 3,1 and Ops 2.1 (`open_port`, `close_port`, `opened_ports`) and 2.7 (`set_ports`).